pkg/lib/saml/service.go

```diff
@@ -227,18 +227,23 @@ func (s *Service) IssueSuccessResponse(
 
 	now := s.Clock.NowUTC()
 	issuerID := s.IdpEntityID()
-	// TODO(saml): Write required fields of the response
 	response := samlprotocol.NewSuccessResponse(now, issuerID, inResponseToAuthnRequest.ID)
 
-	// TODO(saml): Use configured destination if configured
 	destination := callbackURL
+	if sp.Destination != "" {
+		destination = sp.Destination
+	}
 	response.Destination = destination
 
-	// TODO(saml): Use configured recipient if configured
 	recipient := callbackURL
+	if sp.Recipient != "" {
+		recipient = sp.Recipient
+	}
 
-	// TODO(saml): Use configured audience if configured
 	audience := callbackURL
+	if sp.Audience != "" {
+		audience = sp.Audience
+	}
```

Comment:
Should the field in the request override the config? Can you leave a reference to the spec here to document what we should do here?

Reply:
https://github.com/authgear/authgear-server/pull/4513#discussion_r1723097475

Config should override the ACSURL, this is to allow different values in destination, recipient and audience.

Reply:
Which section of the spec specifies this. Include the section here as reference for future reader.

Reply:
Document this behavior is borrowed from another implementation.